lib/db/keyer.go

```diff
@@ -99,6 +105,13 @@ type keyer interface {
 
 	// Block lists
 	GenerateBlockListKey(key []byte, hash []byte) blockListKey
+
+	// Pending (unshared) folders and devices
+	GeneratePendingFolderKey(key, folder, device []byte) (pendingFolderKey, error)
+	FolderFromPendingFolderKey(key []byte) []byte
+	DeviceFromPendingFolderKey(key []byte) ([]byte, bool)
+	GeneratePendingDeviceKey(key, device []byte) pendingDeviceKey
+	DeviceFromPendingDeviceKey(key []byte) []byte
```

Comment:
I'd separate the two with blank line & comment as they have different prefixes.

Reply:
Good idea, will do.